rust/src/array.rs

```diff
@@ -209,10 +215,11 @@ macro_rules! def_primitive_array {
 impl<T: ArrowPrimitiveType> From<ArrayDataRef> for PrimitiveArray<T> {
     fn from(data: ArrayDataRef) -> Self {
         assert!(data.buffers().len() == 1);
-        let values = data.buffers()[0].copy();
+        let raw_values = data.buffers()[0].raw_data();
+        debug_assert!(memory::is_aligned::<u8>(raw_values, mem::size_of::<T>()));
```

Comment:
why is this an debug assert only? I also would like to protect my production code against these kind of errors.

Also, could you test this? There's [`should_panic`](https://doc.rust-lang.org/book/second-edition/ch11-01-writing-tests.html#checking-for-panics-with-should_panic).

Reply:
Very good point. I'll change this to `assert`. For testing though, I need to find a way to bypass `allocate_aligned` and inject some mis-aligned data for `Buffer`. Let me think about it...